packages/eas-json/src/submit/schema.ts

```diff
@@ -34,8 +36,44 @@ export const IosSubmitProfileSchema = Joi.object({
   metadataPath: Joi.string(),
 });
 
-export const SubmitProfileSchema = Joi.object({
+// more strict version after resolving all of the values
+export const ResolvedIosSubmitProfileSchema = Joi.object({
+  ascApiKeyPath: Joi.string(),
+  ascApiKeyId: Joi.string()
+    .regex(/^[\dA-Z]{10}$/)
+    .message(
+      'Invalid Apple App Store Connect API Key ID ("ascApiKeyId") was specified. It should consist of 10 upper case letters or digits. Example: "AB32CDE81F". Learn more: https://expo.fyi/creating-asc-api-key.'
+    ),
+  ascApiKeyIssuerId: Joi.string()
+    .uuid()
+    .message(
+      'Invalid Apple App Store Connect API Key Issuer ID ("ascApiKeyIssuerId") was specified. It should be a valid UUID. Example: "b4d78f58-48c6-4f2c-96cb-94d8cd76970a". Learn more: https://expo.fyi/creating-asc-api-key.'
+    ),
+  appleId: Joi.string()
+    .email()
+    .message(
+      'Invalid Apple ID was specified. It should be a valid email address. Example: "name@domain.com".'
```

Comment:
Prefer to use domains that are guaranteed not to be used in examples. example.com and the .example TLD are valid for examples.
```suggestion
      'Invalid Apple ID was specified. It should be a valid email address. Example: "name@example.com".'
```